src/ompl/tools/lightning/DynamicTimeWarp.h

```diff
@@ -0,0 +1,117 @@
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the JSK, The University of Tokyo nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Dave Coleman
+   Desc:   Compute Dynamic Time Warping
+*/
+
+#ifndef OMPL_TOOLS_LIGHTNING_DYNAMIC_TIME_WARP_
+#define OMPL_TOOLS_LIGHTNING_DYNAMIC_TIME_WARP_
+
+#include <ompl/geometric/PathGeometric.h>
+#include <ompl/base/SpaceInformation.h>
+
+namespace ompl
+{
+namespace tools
+{
+
+namespace og = ompl::geometric;
+namespace ob = ompl::base;
+//namespace ot = ompl::tools;
+
+/// @cond IGNORE
+OMPL_CLASS_FORWARD(DynamicTimeWarp);
+/// @endcond
+
+/** \class ompl::geometric::DynamicTimeWarpPtr
+    \brief A boost shared pointer wrapper for ompl::tools::DynamicTimeWarp */
+
+class DynamicTimeWarp
+{
+public:
+    explicit DynamicTimeWarp(const base::SpaceInformationPtr &si);
+
+    /**
+     * \brief Use Dynamic Timewarping to score two paths
+     * \param path1
+     * \param path2
+     * \return score
+     */
+    double calcDTWDistance(const og::PathGeometric &path1, const og::PathGeometric &path2 ) const;
+
+    /**
+     * \brief Use dynamic time warping to compare the similarity of two paths
+     *        Note: this will not interpolate either of the paths
+     *        Note: before calling this function you might want to reverse one of the paths so that their
+     *        start and goals are property aligned (and match better)
+     * \param path1 - const, will not interpolate
+     * \param path2 - const, will not interpolate
+     * \return score
+     */
+    double getPathsScoreConst(const og::PathGeometric &path1, const og::PathGeometric &path2);
```

Comment:
The comment

```
\param path1 - const, will not interpolate
```

varies between each version of the function. There are some places in the Lightning code that need to already interpolate the path, and some others that do not. Having these extra functions prevents unnecessary interpolation and extra memory copying